projectile.el

```diff
@@ -4527,11 +4527,16 @@ See `projectile--cleanup-known-projects'."
     (message "No projects needed to be removed.")))
 
 ;;;###autoload
-(defun projectile-clear-known-projects ()
-  "Clear both `projectile-known-projects' and `projectile-known-projects-file'."
-  (interactive)
+(defun projectile-clear-known-projects (rediscover)
+  "Clear both `projectile-known-projects' and `projectile-known-projects-file'.
+
+With a prefix argument REDISCOVER rediscovers projects."
+  (interactive "P")
   (setq projectile-known-projects nil)
-  (projectile-save-known-projects))
+  (when rediscover
+    (projectile-save-known-projects)
```

Comment:
Setting the projects to `nil` without serializing this would result in all projects being the same as before next time you run Projectile. When I suggested the `discover` option I meant it to be something that always does auto-discovery, regardless of `projectile-auto-discover`. Alternatively we can have a `projectile-reset-projects` commands, that just always does remove + rediscovery. Maybe this is going to be more intuitive.